### PR TITLE
Added function and test that check login page source response 

### LINKED ISSUE #28: Add login page source code validation to be as expected
Make sure that after response we've got expected source

### PR DESCRIPTION
**Pull request by MaxxxZu**

### CODE REVIEW

**Comment by dkultasev** [comment]
The same here. You can rename it to something like "Add 2 unit tests for ..."

**Comment by dkultasev** [comment]
Actually I'll rephrase myself. For the issue title you can use titles that would only developer understand, but Pull Request is something what you say for end users, they don't know the code  and the function names

File: smarsy/parse.py [hunk 1]
```diff
@@ -86,6 +86,15 @@ def get_headers():
     return open_json_file(file_path)
 
 
+def childs_page_return_right_login(response_page):
+    """
+    Receive HTML page from login function and check we've got expected source
+    """
+    smarsy_login = get_user_credentials()['username']
```
Comment (dkultasev) [inline smarsy/parse.py:93]: username should be passed as a param then to avoid extra dependency (`get_user_credentials`).

Comment (dkultasev) [reply smarsy/parse.py:93]: additionally there are few steps in the function already with single unit test. First step is get user credentials that is not tested. The only step that validates is tested

File: tests/test_parser.py [hunk 2]
```diff
@@ -323,6 +320,19 @@ def test_login_page_has_expected_title(self):
         actual = validate_title(html)
         self.assertTrue(actual)
 
+    @patch('smarsy.parse.get_user_credentials',
+           return_value={'username': 'login'})
+    def test_childs_page_has_expected_username(self, mocked_value):
+        response_string = '\n<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN" \
```
Comment (dkultasev) [inline tests/test_parser.py:326]: there is no need to create such big test strings, 1-2 lines is enough for test

File: smarsy/parse.py (hunk 1 above)
Comment (MaxxxZu) [reply smarsy/parse.py:93]: def get_user_credentials() has no parameters, I call this function and then just take one key 'username'

File: tests/test_parser.py (hunk 2 above)
Comment (MaxxxZu) [reply tests/test_parser.py:326]: OK

File: smarsy/parse.py (hunk 1 above)
Comment (dkultasev) [reply smarsy/parse.py:93]: you need to pass username value to the `childs_page_return_right_login` to get rid of `get_user_credentials` dependency

**Comment by MaxxxZu** [comment]
> you need to pass username value to the `childs_page_return_right_login` to get rid of `get_user_credentials` dependency 

as param in childs_page_return_right_login?

File: smarsy/parse.py (hunk 1 above)
Comment (MaxxxZu) [reply smarsy/parse.py]: https://github.com/dkultasev/smarsy-py/pull/33/commits/272d698b809806c4db8eb9502037b77e3e770e03
Commit added without dependency

File: tests/test_parser.py [hunk 3]
```diff
@@ -323,6 +320,13 @@ def test_login_page_has_expected_title(self):
         actual = validate_title(html)
         self.assertTrue(actual)
 
+    def test_childs_page_has_expected_username(self):
+        response_string = '\n<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN" \
+                     "http://www.w3.org/TR/html4/strict.dtd">\n<HTML>\n \
+                     <HEAD>\n<TITLE>login</TITLE>\n'
+        self.assertTrue(childs_page_return_right_login(response_string,
+                                                       'login'))
+
```
Comment (dkultasev) [inline tests/test_parser.py:330]: there is no negative test. what will happen if we pass invalid value?

Comment (MaxxxZu) [reply tests/test_parser.py:330]: If this way is OK, i add negative test

**Comment by dkultasev** [comment]
and the last thing - increase version

**Review by dkultasev — APPROVED** [review]

**Pull request merged**